brave/src/main/java/brave/http/HttpTracing.java

```diff
@@ -0,0 +1,43 @@
+package brave.http;
+
+import brave.Tracing;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class HttpTracing {
```

Comment:
@llinder @jplock @hyleung @johnament @takezoe @devinsba so I think this is the latest thing to use in injection/configuraion. HttpTracing includes everything, including the normal things in the Tracing component (like the tracer and propagation), and the replacement for SpanNameProvider.

HttpTracing is nice because it should work nicely in any dependency injection system, like Spring, Guice or CDI, enclosing the non-instrumentation specific tracing configuration.

I haven't doc'd this class yet, but I would just note that the client and server parsers are where you do things like add tags based on servlet principal or http headers